src/transaction/validation.rs

```diff
@@ -226,7 +253,7 @@ impl Transaction {
         let duplicated_utxo_id = self
             .inputs()
             .iter()
-            .filter_map(|i| i.is_coin().then(|| i.utxo_id()));
+            .filter_map(|i| i.is_coin().then(|| i.utxo_id()).flatten());
 
```

Comment:
Do we need to check for duplicate message inputs separately here since messages don't have a UTXO ID?

Reply:
The output of `Input::utxo_id` is now `Option<&UtxoId>` instead of `&UtxoId`. This means the check will be executed only for inputs that contains utxo id